Competition2017/src/competition/subsystems/BaseXCANTalonSpeedControlledSubsystem.java

```diff
@@ -0,0 +1,202 @@
+        
+        followerMotor = factory.getCANTalonSpeedController(followChannel);
+        initializeFollowerMotorConfiguration(invertFollower);
+    }
+    
+    /**
+     * 
+     * @param name What the system is called. This will apply to various Properties.
+     * @param masterChannel The CAN index of the master motor (or the only motor, for a simple system)
+     * @param factory The WPIFactory
+     * @param pidPropertyManager The default PIDF values the system should use
+     * @param propManager The XPropertyManager
+     */
+    public BaseXCANTalonSpeedControlledSubsystem(
+            String name,
+            int masterChannel,
+            boolean invertMaster,
+            boolean invertMasterSensor,
+            WPIFactory factory, 
+            PIDPropertyManager pidPropertyManager,
+            XPropertyManager propManager){
+        super(name);
+        log.info("Creating " + name + " system");
+        
+        this.pidPropertyManager = pidPropertyManager;
+        this.masterChannel = masterChannel;
+        
+        // Reading through the manual, it looks like only a few things 
+        // support human units. It might be easier to just keep everything in native units per 100ms,
+        // which is what the Talon uses for all its calculations.
+        // I call this Ticks per Deciseconds, or TPD
+        systemSpeedThresh = propManager.createPersistentProperty(name + " nominal speed thresh (TPC)", 1);
+        systemCurrentSpeed = propManager.createEphemeralProperty(name + " current speed (TPC)", 0);
+        systemTargetSpeed = propManager.createEphemeralProperty(name + " goal speed (TPC)", 0);
+        systemOutputPower = propManager.createEphemeralProperty(name + " voltage", 0);
+        atSpeedProp = propManager.createEphemeralProperty("Is" + name + " at speed?", false);
+        systemTalonError = propManager.createEphemeralProperty(name + " speed error", 0);
+        enablesystemLogging = propManager.createEphemeralProperty("Is " + name + " logging enabled?", false);
+        
+        masterMotor = factory.getCANTalonSpeedController(masterChannel);
+        initializeMasterMotorConfiguration(invertMaster, invertMasterSensor);
+        masterMotor.createTelemetryProperties(name + "  master", propManager);
+    }
+    
+    protected void initializeMasterMotorConfiguration(boolean motorInverted, boolean motorSensorInverted) {
+        masterMotor.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        masterMotor.setInverted(motorInverted);
+        masterMotor.reverseSensor(motorSensorInverted);
+        masterMotor.setControlMode(TalonControlMode.Speed);
+        masterMotor.setProfile(0);
+    }
+    
+    protected void initializeFollowerMotorConfiguration(boolean motorInverted) {
+        if (followerMotor == null) {
+            log.warn("initializeFollowerMotorConfiguration was called, but no followerIndex was given during creation!");
```

Comment:
Can this class be made into two -- a single-motor version and a subclass which operates on a pair? The interface would be the same as this except that you wouldn't need these hacky checks.